src/types.jl

```diff
@@ -282,24 +282,10 @@ end
 
 # Equality
 ==(a::ZonedDateTime, b::ZonedDateTime) = a.utc_datetime == b.utc_datetime
+isequal(a::ZonedDateTime, b::ZonedDateTime) = isequal(a.utc_datetime, b.utc_datetime)
+hash(zdt::ZonedDateTime, h::UInt) = hash(zdt.utc_datetime, hash(:ZonedDateTime, h))
```

Comment:
I think we should just use a hash seed which is stored as a global constant. That way other types which want to hash to the same value can reference the same seed. This would be similar to how `Range` hashing works (`hash(1:2, zero(UInt))`)

```julia
const UTC_INSTANT_HASH_SEED = UInt === UInt64 ? 0xb048e5784b3097fa : 0x2e77bd2b
function hash(zdt::ZonedDateTime, h::UInt)
    h += UTC_INSTANT_HASH_SEED
    h = hash(zdt.utc_datetime, h)  
end
```

Reply:
I'm not sure what the advantage of that is over just using a hash of the type name? I feel like using the type name is more readable at least.

Reply:
The main advantage to this approach is that the UTC instant value isn't tied to any specific type so we shouldn't tie the seed to any specific name. Also, since the seed is a global it's easy to reference by other packages.

Reply:
Might as well hash a literal string "tz" or something, rather than a constant. Easier for other packages to use that.

Reply:
If we want to hash a `Symbol` or `String` as the seed then I would vote for `:utc_instant`

Reply:
> UTC instant value isn't tied to any specific type

Except that it kind of is. It's tied to some kind of localized or zoned DateTime.